src/MongoDB.Bson/BsonDefaults.cs

```diff
@@ -27,11 +28,28 @@ public static class BsonDefaults
         private static bool __dynamicArraySerializerWasSet;
         private static IBsonSerializer __dynamicArraySerializer;
         private static bool __dynamicDocumentSerializerWasSet;
-        private static IBsonSerializer __dynamicDocumentSerializer; 
+        private static IBsonSerializer __dynamicDocumentSerializer;
         private static GuidRepresentation __guidRepresentation = GuidRepresentation.CSharpLegacy;
+        private static GuidRepresentationMode __guidRepresentationMode = GuidRepresentationMode.V2;
         private static int __maxDocumentSize = int.MaxValue;
         private static int __maxSerializationDepth = 100;
 
+        // static constructor
+        static BsonDefaults()
+        {
+            var testWithDefaultGuidRepresentation = Environment.GetEnvironmentVariable("TESTWITHDEFAULTGUIDREPRESENTATION");
+            if (testWithDefaultGuidRepresentation != null)
+            {
+                var _ = Enum.TryParse(testWithDefaultGuidRepresentation, out __guidRepresentation); // ignore errors
+            }
+
+            var testWithDefaultGuidRepresentationMode = Environment.GetEnvironmentVariable("TESTWITHDEFAULTGUIDREPRESENTATIONMODE");
+            if (testWithDefaultGuidRepresentationMode != null)
+            {
+                var _ = Enum.TryParse(testWithDefaultGuidRepresentationMode, out __guidRepresentationMode); // ignore errors
+            }
+        }
```

Comment:
This change lets me inject different defaults via environment variables for testing.

Reply:
Several notes:
1. `AssemblyFixture` is implemented based on the xunit examples here: https://github.com/xunit/samples.xunit/tree/master/AssemblyFixtureExample (the implementation is almost the same but with really small refactoring)
2. The implementation looks quite big, but the real changes (comparing with the currently used) are really small. They are:
   -  https://github.com/DmitryLukyanov/mongo-csharp-driver/commit/7a2d57d92a4e1b13b64cd6a3891598acb787bd01#diff-add1fdc6275946baface6fed31e822a8R87-R111  where we just create and save the assembly fixtures to pass them to the lower levels.
   - https://github.com/DmitryLukyanov/mongo-csharp-driver/commit/7a2d57d92a4e1b13b64cd6a3891598acb787bd01#diff-add1fdc6275946baface6fed31e822a8R117-R120 where we release resource (if available)
   - https://github.com/DmitryLukyanov/mongo-csharp-driver/commit/7a2d57d92a4e1b13b64cd6a3891598acb787bd01#diff-add1fdc6275946baface6fed31e822a8R171-R175
In these lines, we just overwrite the possible collection fixtures if they exist (or create a new one). So, in two words, the current suggestion adds collection fixtures globally, unlike usual CollectionFixtures which are added on-demand

Reply:
NOTE: `Aggregator.Run` is just `try/catch` logic

Reply:
Also, I would suggest considering to enable parallel test calls in some cases (currently we've disabled parallel run at all) regarding this article: https://xunit.net/docs/running-tests-in-parallel.html. 
It looks like we can mark the test classes which we want to run without parallel calls by `Collection` attribute. Then all other tests which can be launched independently will be able to be launched in parallel. I guess it will improve tests performance a little bit.
I'm happy to create a ticket about it.

Reply:
Ack